tests/c_api_tests/test_cpp_api_serialization.cpp

```diff
@@ -130,7 +130,7 @@ TEST_CASE("C++ API Serialization and Deserialization") {
         auto check_deserializer = [&](Deserializer& deserializer) {
             // get dataset
             auto& dataset = deserializer.get_dataset();
-            auto& info = dataset.get_info();
+            auto const& info = dataset.get_info();
```

Comment:
Commenting on this line for reference. Shouldn't we test that we can `add_attribute_buffer` to `writable_dataset` when generated by the `deserializer`? Or is it enough with the unit tests that cover the (de)serializer? I think it is a good idea that we add a specific test for this here since it was introduced both in the C and Cpp APIs.

Reply:
You cannot test `set_attribute_buffer` separately. There is no way to go to the opaque struct to see if the attribute buffer is actually set.

You can only test this function together with de-serializer to see if the columnar data are parsed and written into the buffer.

Reply:
That is what I meant, hence why I commented on the serialization test file. Isn't this still needed? Because else the C and Cpp API call of `set_attribute_buffer` isn't being tested.

Reply:
Ah, now I get it. 

@Jerry-Jinfeng-Guo, indeed we need to create a columnar version of test de-serializer in this test case, to test the `set_attribute_buffer` C-API function.

Reply:
Added in the newest commit.